src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -108,14 +108,18 @@ SMALL_RECT Terminal::_GetSelectionRow(const SHORT row, const COORD higherCoord,
 }
 
 // Method Description:
-// - Get the current anchor position
+// - Get the current anchor position relative to the whole text buffer
 // Arguments:
 // - None
 // Return Value:
 // - None
 const COORD Terminal::GetSelectionAnchor() const
 {
-    return _selectionAnchor;
+    COORD selectionAnchorPos;
+    selectionAnchorPos.X = _selectionAnchor.X;
+    selectionAnchorPos.Y = _selectionAnchor.Y + gsl::narrow<short>(ViewStartIndex());

```

Comment:
```suggestion
    THROW_IF_FAILED(ShortAdd(selectionAnchorPos.Y, static_cast<SHORT>(ViewStartIndex()), &selectionAnchorPos.Y));
```
also, make sure you're updating/using `_selectionVerticalOffset` appropriately. This var should be updated every time we are setting a selection anchor.

Reply:
I think I can use _selectionVerticalOffset here. 

Actually, this method only returns the calculated selection position, it does not change anything :)